doc/link_cut_trees.html

```diff
@@ -0,0 +1,209 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
+
+<html>
+<head>
+  <meta http-equiv="Content-Language" content="en-us">
+  <meta http-equiv="Content-Type" content="text/html; charset=us-ascii">
+
+  <title>Boost Link/Cut Trees</title>
+</head>
+
+<body bgcolor="#FFFFFF" link="#0000EE" text="#000000" vlink="#551A8B" alink=
+"#FF0000">
+  <img src="../../../boost.png" alt="C++ Boost" width="277" height=
+  "86"><br clear="none">
+
+  <h1><a name="sec:link-cut-trees" id="sec:link-cut-trees"></a> Link/Cut Trees</h1>
+  <pre>link_cut_trees&lt;ElementParentMap, ElementChildMap&gt;</pre>
+
+  <p>A link/cut-trees data structure
+  maintains a forest of element nodes subject to dynamic linking and cutting operations.
+  Rooted trees are encoded in both the <tt>ElementParentMap</tt> and <tt>ElementChildMap</tt>
+  property maps. Splay trees are used internally to represent every link/cut-tree [<a href="bibliography.html#tarjan83">77</a>]..</p>
+
+  <h3>Where Defined</h3><a href=
+  "../../../boost/graph/link_cut_trees.hpp"><tt>boost/graph/link_cut_trees.hpp</tt></a>
+
+  <h3>Template Parameters</h3>
+
+  <table border summary="">
+    <tr>
+      <td><tt>ElementParentMap</tt></td>
+
+      <td>Must be a model of <a href=
+      "../../property_map/doc/ReadWritePropertyMap.html">ReadWritePropertyMap</a>
+      and the key and value type the same as the trees' element type.</td>
+    </tr>
+
+    <tr>
+      <td><tt>ElementChildMap</tt></td>
+
+      <td>Default the same as <tt>ElementParentMap</tt>. Also must be a model of <a href=
+      "../../property_map/doc/ReadWritePropertyMap.html">ReadWritePropertyMap</a>
+      and the key and value type the same as the trees' element type.</td>
+    </tr>
+  </table>
+
+  <h3>Example</h3>
+
+  <p>A typical usage pattern for <tt>link_cut_trees</tt> can be seen in the
+  dynamic connectivity problem for acyclic graphs. Given two nodes <tt>x</tt> and <tt>y</tt>, 
+  they are connected if and only if <tt>find_root(x) == find_root(y)</tt>.</p>
+
+  <pre>
+  ...
+  link_cut_trees&lt;ElementParentMap, ElementChildMap&gt; lct(p, l, r);
```

Comment:
Where do those parameters--p, l, r--come from? I might be missing it, but I can't see them defined earlier.

Reply:
This code snippet is a tiny example showing the basic usage, with variables declaration/initiation omitted. Runnable examples can be found in `test/link_cut_trees_test.cpp`.
With that being said, I've updated their names to improve readability.